pertpy/tools/_pert_sign.py

```diff
@@ -0,0 +1,112 @@
+from typing import Optional
+
+import numpy as np
+import scanpy as sc
+from anndata import AnnData
+from pynndescent import NNDescent
+from scanpy.tools._utils import _choose_representation
+from scipy.sparse import csr_matrix, issparse
+
+
+def pert_sign(
+    adata: AnnData,
+    pert_key: str,
+    control: str,
+    split_by: Optional[str] = None,
+    n_neighbors: int = 20,
+    use_rep: Optional[str] = None,
+    n_pcs: Optional[int] = None,
+    batch_size: Optional[int] = None,
+    copy: bool = False,
+    **kwargs
+):
+    """Calculate perturbation signature.
+
+    For each cell, we identify `n_neighbors` cells from the control pool with
+    the most similar mRNA expression profiles. The perturbation signature is calculated by subtracting
+    the averaged mRNA expression profile of the control neighbors from the mRNA expression profile
+    of each cell.
+
+    Args:
+        adata: The annotated data object.
+        pert_key: The column  of `.obs` with perturbation categories, should also contain `control`.
+        control: Control category from the `pert_key` column.
+        split_by: Provide the column `.obs` if multiple biological replicates exist to calculate
+            the perturbation signature for every replicate separately.
```

Comment:
I think that the words in the new line have to match the indentation in the line above or Sphinx will complain.

-> 

```
  split_by: Provide the column `.obs` if multiple biological replicates exist to calculate
                the perturbation signature for every replicate separately.
```

Reply:
Hm, I followed the examples from [the link](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).

Reply:
All right. My memory might be off. I'll happily test it as soon as you've added it to the usage.md. I can also do it later myself, no problem.